provider/bikeshed/src/units/essentials.rs

```diff
@@ -18,6 +19,57 @@ use zerovec::ZeroMap;
 
 impl DataProvider<UnitsEssentialsV1Marker> for DatagenProvider {
     fn load(&self, req: DataRequest) -> Result<DataResponse<UnitsEssentialsV1Marker>, DataError> {
+        /// Fills the prefixes map with the SI prefixes (binary and decimal)
+        ///
+        /// # Errors
+        ///
+        /// - `DataError::custom("Invalid prefix: must be Binary or Decimal")` if the prefix is not Binary or Decimal
+        /// - `DataError::custom("Failed to parse pattern key value")` if the pattern key value cannot be parsed
+        fn fill_si(
+            data: &BTreeMap<String, Patterns>,
+            prefixes_map: &mut BTreeMap<PatternKey, String>,
+        ) -> Result<(), DataError> {
+            const BINARY_PREFIX: &str = "1024p";
+            const DECIMAL_PREFIX: &str = "10p";
+
+            data.iter()
+                .filter(|(key, _)| {
+                    key.starts_with(BINARY_PREFIX) || key.starts_with(DECIMAL_PREFIX)
+                })
+                .map(|(key, patterns)| {
+                    let pattern_key_result =
+                        if let Some(trimmed_key) = key.strip_prefix(BINARY_PREFIX) {
+                            trimmed_key.parse::<u8>().map(PatternKey::Binary)
+                        } else if let Some(trimmed_key) = key.strip_prefix(DECIMAL_PREFIX) {
+                            trimmed_key.parse::<i8>().map(PatternKey::Decimal)
+                        } else {
+                            return Err(DataError::custom(
+                                "Invalid prefix: must be Binary or Decimal",
+                            )
+                            .with_debug_context(key));
```

Comment:
This is dead code, you probably should remove the `filter` call

Reply:
I added in case there are more prefix added to the filter, this will return an error

Reply:
It will never be reached because of the `filter`

Reply:
if the filter has more than these two cases  in the filter `key.starts_with(BINARY_PREFIX) || key.starts_with(DECIMAL_PREFIX)` , it will hit this case

Reply:
but the filter doesn't

Reply:
yes, for now, we can also add `unreachable!`, but we need to return an error if this happen

what is your suggestion ?

Reply:
So currently your code will not fail for CLDR data that has a prefix other than decimal or binary. I think it should fail, because datagen cannot handle that input. You can make it fail (with the `DataError` that you have already written) by removing the filter.